src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+      * Case 1.b:   viewAngles[1] < 180 degrees         => Sphere - (Cone + Cone) (Cones on z axis expanding from origin)
+
+    * Case 2:       viewAngles[0] = 180 degrees
+
+      * Case 2.a:   viewAngles[1] = 180 degrees         => Hemisphere View Region
+      * Case 2.b:   viewAngles[1] < 180 degrees         => Hemisphere - (Cone + Cone) (Cones on appropriate hemispheres)
+
+    * Case 3:       viewAngles[1] = 180 degrees
+
+      * Case 3.a:   viewAngles[0] < 180 degrees         => Sphere intersected with Pyramid View Region
+      * Case 3.b:   viewAngles[0] > 180 degrees         => Sphere - Backwards Pyramid View Region
+
+    * Case 4:       viewAngles[0 & 1] < 180             => Capped Pyramid View Region
+    * Case 5:       viewAngles[0] > 180, Altitude < 180 => (Sphere - (Cone + Cone) (Cones on appropriate hemispheres)) - Backwards Capped Pyramid View Region
+
+    Args:
+        visibleDistance: The view distance for this region.
+        viewAngles: The view angles for this region.
+        name: An optional name to help with debugging.
+        position: An optional position, which determines where the center of the region will be.
+        rotation: An optional Orientation object which determines the rotation of the object in space.
+        orientation: An optional vector field describing the preferred orientation at every point in
+          the region.
+        tolerance: Tolerance for collision computations.
+    """
+    def __init__(self, visibleDistance, viewAngles, name=None, position=Vector(0,0,0), rotation=None,\
+        orientation=None, tolerance=1e-8):
+        # Bound viewAngles from either side.
+        viewAngles = tuple([min(angle, math.tau) for angle in viewAngles])
+
+        if min(viewAngles) <= 0:
+            raise ValueError("viewAngles cannot have a component less than or equal to 0")
+
+        # Cases in view region computation
+        # Case 1:       Azimuth view angle = 360 degrees
+        #   Case 1.a:   Altitude view angle = 180 degrees   => Full Sphere View Region
+        #   Case 1.b:   Altitude view angle < 180 degrees   => Sphere - (Cone + Cone) (Cones on z axis expanding from origin)
+        # Case 2:       Azimuth view angle = 180 degrees
+        #   Case 2.a:   Altitude view angle = 180 degrees   => Hemisphere View Region
+        #   Case 2.b:   Altitude view angle < 180 degrees   => Hemisphere - (Cone + Cone) (Cones on appropriate hemispheres)
+        # Case 3:       Altitude view angle = 180 degrees   
+        #   Case 3.a:   Azimuth view angle < 180 degrees    => Sphere intersected with Pyramid View Region
+        #   Case 3.b:   Azimuth view angle > 180 degrees    => Sphere - Backwards Pyramid View Region 
+        # Case 4:       Both view angles < 180              => Capped Pyramid View Region
+        # Case 5:       Azimuth > 180, Altitude < 180       => (Sphere - (Cone + Cone) (Cones on appropriate hemispheres)) - Backwards Capped Pyramid View Region
```

Comment:
Can you combine cases 1.b, 2.b, 4, and 5, i.e., all cases where `viewAngles[1] < 180 degrees`? It seems like in all of those you can take the whole sphere, chop out the wedge in the back excluded by `viewAngles[0]`, and then subtract the two cones. I think the remaining cases (`viewAngles[1] = 180 degrees`) are the same except you don't use the cones. But maybe I'm confused?

Reply:
We could, but to be honest getting these results consistent was already really finicky and depends heavily on the order of operations and choice of engine. I'd rather not mess with it unless there's a good reason.

Reply:
OK, let's just put it on the back burner list then.